tensorflow/python/ops/numpy_ops/np_array_ops.py

```diff
@@ -560,6 +561,23 @@ def _reduce(tf_fn,
       tf_fn(input_tensor=a.data, axis=axis, keepdims=keepdims))
 
 
+# TODO (DarrenZhang01): Add `axis` support to the `size` API.
+@np_utils.np_doc('size')
+def size(x, axis=None):   # pylint: disable=missing-docstring
+  x = asarray(x).data
+  if axis is not None:
+    raise NotImplementedError("axis argument is not supported in the current "
+                              "`np.size` implementation")
+  if isinstance(x, (int, float, np.int32, np.int64,
```

Comment:
After your change above, this should not happen. `x` will **always** be a Tensor. You can move this above the `asarray` call, since I think it makes sense as an optimization.

Reply:
Great idea!